jwst/cube_build/blot_cube_build.py

```diff
@@ -161,145 +166,185 @@ def blot_images(self):
-                d3_matrix = np.tile(d3 * d3, [dxy_matrix.shape[0], 1])
-
-                wdistance = dxy_matrix + d3_matrix
-                weight_distance = np.power(np.sqrt(wdistance), self.weight_power)
-                weight_distance[weight_distance < lower_limit] = lower_limit
-                weight_distance = 1.0 / weight_distance
-
-                # determine the spaxel xx_cube,yy_cube values of these spaxels
-                # in the ROI so they can be used to pull out the flux of the
-                # median sky cube.
-                yy_cube = (indexr[0] / self.naxis1).astype(np.int)
-                xx_cube = indexr[0] - yy_cube * self.naxis1
-                scf = np.array([self.cube_flux[zz, yy_cube[ir], xx_cube[ir]]
-                                for ir, rr in enumerate(indexr[0]) for zz in indexz[0]])
-                scf = np.reshape(scf, weight_distance.shape)
-                blot_flux[yy, xx] = np.sum(weight_distance * scf)
+                dxy = np.sqrt(dxy)
+                weight_distance = np.exp(-dxy)
+                blot_flux[ydet[ipt], xdet[ipt]] = np.sum(weight_distance * flux_cube[indexr])
                 blot_weight = np.sum(weight_distance)
 
                 # check for blot_weight !=0
                 if blot_weight == 0:
-                    blot_flux[yy, xx] = 0
+                    blot_flux[ydet[ipt], xdet[ipt]] = 0
                 else:
-                    blot_flux[yy, xx] = blot_flux[yy, xx] / blot_weight
-# ________________________________________________________________________________
-            blot.data = blot_flux
-            blot_models.append(blot)
-        t1 = time.time()
-        log.info("Time Blot images = %.1f.s" % (t1 - t0,))
-        return blot_models
+                    blot_flux[ydet[ipt], xdet[ipt]] =  \
+                        (blot_flux[ydet[ipt], xdet[ipt]] / blot_weight)
+
+        #  after all looping
+        # t1 = time.time()
+        # print('Time to call blot_overlap',t1 - t0)
+        return blot_flux
+    # ________________________________________________________________________________
+
+    def blot_overlap_quick(self, xcenter, ycenter, x_cube, y_cube,
+                           flux_cube, blot_flux):
+
+        # blot_overlap_quick finds to overlap between the blotted sky values (x_cube, y_cube)
+        # and the detector pixels. This is faster than blot_overlap.
+        # Looping is done over irregular arrays (x_cube, y_cube) and mapping to xcenter, ycenter
+        # is quicker than looping over detector x,y and finding overlap with large array for
+        # x_cube, y_cube
+        # t0 = time.time()
+        blot_weight = np.zeros_like(blot_flux)
+        num = x_cube.size
+
+        # print('blot_overlap_quick Number of elements on sky mapped to detector going to loop over',num)
+        # TODO need to move this value (roi_det)  to reference file  or at least in spec ?
+        roi_det = 0.5  # 1/2 a pixel
+        for ipt in range(0, num - 1):
```

Comment:
It looks as if you're skipping the last element in `x_cube` and `y_cube`.  Was that intentional?